javascript/ql/src/semmle/javascript/dataflow/internal/InterModuleTypeInference.qll

```diff
@@ -248,10 +248,18 @@ private class AnalyzedAmdImport extends AnalyzedPropertyRead, DataFlow::Node {
   Module required;
 
   AnalyzedAmdImport() {
-    exists(AmdModule amd, PathExpr dep, Parameter p |
-      amd.getDefine().dependencyParameter(dep, p) and
-      this = DataFlow::parameterNode(p) and
-      required.getFile() = amd.resolve(dep)
+    exists(AmdModule amd, PathExpr dep |
+      exists(Parameter p |
+        amd.getDefine().dependencyParameter(dep, p) and
+        this = DataFlow::parameterNode(p)
+      )
+      or
+      exists(CallExpr requireCall |
```

Comment:
Could you add a type inference test that uses AMD `require`?

Reply:
Sure, I can do that.

Reply:
I've amended the commit to include this test.